examples/camera/camera_orbit.rs

```diff
@@ -111,40 +131,36 @@ fn instructions(mut commands: Commands) {
 fn orbit(
     mut camera: Query<&mut Transform, With<Camera>>,
     camera_settings: Res<CameraSettings>,
-    keyboard_input: Res<ButtonInput<KeyCode>>,
-    time: Res<Time>,
+    mut mouse_buttons: EventReader<MouseButtonInput>,
+    mouse_motion: Res<AccumulatedMouseMotion>,
+    mut pressed: ResMut<MouseButtonsPressed>,
 ) {
     let mut transform = camera.single_mut();
-
-    let mut delta_pitch = 0.0;
+    let delta = mouse_motion.delta;
     let mut delta_roll = 0.0;
-    let mut delta_yaw = 0.0;
 
-    if keyboard_input.pressed(KeyCode::KeyW) {
-        delta_pitch += camera_settings.orbit_speed;
-    }
-    if keyboard_input.pressed(KeyCode::KeyS) {
-        delta_pitch -= camera_settings.orbit_speed;
-    }
-
-    if keyboard_input.pressed(KeyCode::KeyQ) {
-        delta_roll -= camera_settings.orbit_speed;
-    }
-    if keyboard_input.pressed(KeyCode::KeyE) {
-        delta_roll += camera_settings.orbit_speed;
+    for button_event in mouse_buttons.read() {
+        if button_event.button == MouseButton::Left {
+            pressed.left = button_event.state.is_pressed();
+        }
+        if button_event.button == MouseButton::Right {
+            pressed.right = button_event.state.is_pressed();
+        }
     }
 
-    if keyboard_input.pressed(KeyCode::KeyA) {
-        delta_yaw -= camera_settings.orbit_speed;
+    if pressed.left {
+        delta_roll -= 1.0;
     }
-    if keyboard_input.pressed(KeyCode::KeyD) {
-        delta_yaw += camera_settings.orbit_speed;
+    if pressed.right {
+        delta_roll += 1.0;
     }
 
-    // Incorporating the delta time between calls prevents this from being framerate-bound.
-    delta_pitch *= time.delta_seconds();
-    delta_roll *= time.delta_seconds();
-    delta_yaw *= time.delta_seconds();
+    // Mouse motion is one of the few inputs that should not be multiplied by delta time,
+    // as we are already receiving the full movement since the last frame was rendered. Multiplying
+    // by delta time here would make the movement slower that it should be.
+    let delta_pitch = delta.y * camera_settings.pitch_speed;
+    let delta_yaw = delta.x * camera_settings.yaw_speed;
+    delta_roll *= camera_settings.roll_speed;
```

Comment:
If you use the mouse button for roll, you need delta t in this line, as the argumentation for mouse movement does not hold for mouse buttons. You can't have clicked the mouse button more than once in a frame per bevy's rules :)

Reply:
I went back and forward TWICE on this :laughing: ah well, good to know for sure.

Reply:
Yeah, delta t is notoriously tricky to get right :D